src/components/cover/helper.ts

```diff
@@ -46,41 +66,81 @@ export const getCoverUrl = ({
   const firstCover = first(covers);
 
   // If no best representation has been given use first cover if available.
-  if (!bestRepresentation && firstCover && getUrl(firstCover, size)) {
-    return getUrl(firstCover, size);
+  if (!bestRepresentation && firstCover && getUrl(firstCover.cover, size)) {
+    return getUrl(firstCover.cover, size);
   }
 
   // See if we can find a cover that has same id as the best representation id.
   const bestRepresentationCover = first(
-    covers.filter(
-      (cover: CoverType) =>
-        bestRepresentation && cover.id === bestRepresentation.pid
-    )
+    covers.filter((item) => bestRepresentation?.pid === item.pid)
   );
-
   // If we have a best representation cover in the given size use that.
-  if (bestRepresentationCover && getUrl(bestRepresentationCover, size)) {
-    return getUrl(bestRepresentationCover, size);
+  if (bestRepresentationCover && getUrl(bestRepresentationCover.cover, size)) {
+    return getUrl(bestRepresentationCover.cover, size);
   }
 
   // If the best representation method failed we try the first cover.
-  if (firstCover && getUrl(firstCover, size)) {
-    return getUrl(firstCover, size);
+  if (firstCover && getUrl(firstCover.cover, size)) {
+    return getUrl(firstCover.cover, size);
   }
 
   // Everything else failed. We don't know what to do 🤷.
   return null;
 };
 
+/**
+ * Filters out null entries from a list of manifestations.
+ * Use when GraphQL returns `(Manifestation | null)[]`.
+ */
+export function filterNonNullManifestations<T>(
+  manifestations: (T | null | undefined)[] | null | undefined
+): T[] {
+  return manifestations?.filter((m): m is T => m != null) ?? [];
+}
+export const createIsbnCql = (ids: (string | number)[] = []) =>
+  ids.length ? ids.map((id) => `term.isbn=${id}`).join(" OR ") : "";
+
+/**
+ * Resolves the list of PIDs to use when fetching covers.
+ * - For `pid`, returns the raw `ids` array.
+ * - For `isbn`, returns an array with the resolved PID if available.
+ */
+export function resolveCoverPids({
+  idType,
+  ids,
+  resolvedIsbnPid
+}: {
+  idType: CoverIdType;
+  ids?: (Pid | string)[];
+  resolvedIsbnPid?: string;
+}): string[] {
```

Comment:
Since this function resolves pids then how about returning `Pid[]` instead of the generic `string[]`?

Reply:
Let's leave it as is.